### PR TITLE
Trailing comma in JSON not allowed

### PR DESCRIPTION
**Pull request by Zbysekz**
Small fix of #866
JSON specification doesn't allow trailing comma.
Was causing exception when i try to`json.loads()` it.

I added test for this case.

[Stackoverflow link](https://stackoverflow.com/questions/201782/can-you-use-a-trailing-comma-in-a-json-object)

Thanks for this again @dkeeney ! Works nicely 👍 

Removed semicolons & removed trailing commas

### CODE REVIEW

**Comment by breznak** [comment]
@Zbysekz looks good, thanks! But some hardcoded cpp tests that compare jsons fail 
https://github.com/htm-community/htm.core/pull/867/checks?check_run_id=953137257#step:5:756

**Comment by Zbysekz** [comment]
ok, it should be fine now. 

But let's wait for @dkeeney if he is fine with this (not sure if it has some impact elsewhere, but probably not)

**Comment by dkeeney** [comment]
Good catch,
In many places JSON implementations allow the trailing comma on the last line even though it is not spec, but some do not so we should remove them. I will try to make sure you got all of them.

**Comment by Zbysekz** [comment]
> In many places JSON implementations allow the trailing comma on the last line even though it is not spec, but some do not so we should remove them. I will try to make sure you got all of them.

Yeah, i think i can handle that, it's on many places but thanks to unit tests its good... wish i had faster machine for build :)

**Comment by dkeeney** [comment]
So fix the test expected results in DatabaseRegion_Test and I will approve it again.

File: py/tests/networkapi/getParameters_test.py [hunk 1]
```diff
@@ -121,11 +121,11 @@ def testGetSpec(self):
       "count": 0,
       "regionLevel": 1,
       "isDefaultOutput": 1
-    },
+    }
   }
 }"""
     net = Network()
-    json_str = net.getSpecJSON("RDSEEncoderRegion");
```
[the remark this answers is not in the document]
Comment (Zbysekz) [reply py/tests/networkapi/getParameters_test.py:128]: yes i have same habit with the semicolons from C++ :)

[the remark this answers is not in the document]
Comment (Zbysekz) [reply py/tests/networkapi/getParameters_test.py:128]: its sometimes hard to get rid of that habit

**Comment by Zbysekz** [comment]
Hmm i have a question.. TEST(DateEncoderRegionTest, testLinking) fails on my PC (ubuntu) with this output:
```
/home/osboxes/HTM/htm.core/src/test/unit/regions/DateEncoderRegionTest.cpp:224: Failure
Value of: r2OutputArray.getSDR() == expected2
  Actual: false
Expected: true
Expected SDR( 34 ) 0, 1, 4, 5, 6, 32, 33
  Found: [ SDR( 34 ) 0, 1, 4, 5, 31, 32, 33
 ]
```

Do you guys know, why is this test OS dependant? (according to the comments in test)

**Comment by breznak** [comment]
> Do you guys know, why is this test OS dependant? (according to the comments in test) 

I'm not sure why it's (now?) failing for you, but an encoder shouldn't certainly be OS dependant. 
Do you assume it is by the comments? I just quickly went through it, imho the `// OSx ...` (and linux) comments are old/wrong. The test is normally run on all CI/platforms with the same results.  

Btw, do your cpp,py tests for `datetime encoder` (not region) pass?

**Comment by Zbysekz** [comment]
this PR seems to be OK now, can be merged

**Comment by Zbysekz** [comment]
I can't merge since i probably don't have permissions for that (because of that incidient before) 
Maybe you give me some rights again, and i will really take care that i will not push some bad stuff.

**Comment by breznak** [comment]
oh, I'll have a look at the rights when i get home

**Pull request merged**